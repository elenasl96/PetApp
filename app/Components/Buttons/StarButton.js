import React from "react";
import { AntDesign } from "@expo/vector-icons";
import { TouchableOpacity } from "react-native";
import dbPlace from "../../firebase/Database/Functions/dbPlace";
import { AuthContext } from "../Custom/AuthContext";

class StarButton extends React.Component {
  static contextType = AuthContext;

  state = {
    favorite: false,
    mounted: false,
  };

  constructor(props) {
    super(props);
    dbPlace.getSavedPlaces(this.props.uid).then((savedPlaces) => {
      savedPlaces.forEach((savedPlaceId) => {
        if (this.props.pid == savedPlaceId) {
          this.setState({ favorite: true });
        }
      });
    });
  }

  componentDidMount() {
    this.setState({ mounted: true });
  }

  componentWillUnmount() {
    this.setState({ mounted: false });
  }

  addToFavorite = () => {
    dbPlace.addSavedPlace(this.props.uid, this.props.pid);
    this.setState({ favorite: true });
    this.context.addFavouritePlace(this.props.pid);
  };

  deleteFromFavorite = () => {
    dbPlace.deleteSavedPlace(this.props.uid, this.props.pid).then(() => {
      this.setState({ favorite: false });
      this.context.deleteFavouritePlace(this.props.pid);
    });
  };

  render() {
    if (this.state.favorite) {
      return (
        <TouchableOpacity
          testID="StarButton.deleteFromFavorite"
          onPress={() => this.deleteFromFavorite()}
          
        >
          <AntDesign
            name="star"
            size={32}
            style={{ marginLeft: 10 }}
            color="orange"
          />
        </TouchableOpacity>
      );
    } else {
      return (
        <TouchableOpacity
          testID="StarButton.addFavorite"
          id="button_fav"
          onPress={() => this.addToFavorite()}
        >
          <AntDesign
            name="staro"
            size={32}
            style={{ marginLeft: 10 }}
            color="orange"
          />
        </TouchableOpacity>
      );
    }
  }
}

export default StarButton;
