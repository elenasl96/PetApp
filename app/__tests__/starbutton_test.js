import React from 'react';
import StarButton from '../Components/Buttons/StarButton';
import Adapter from 'enzyme-adapter-react-16';
import {shallow,mount,configure} from 'enzyme';

let wrapper;
let mockAddFavorite;

configure({adapter: new Adapter()});

beforeEach(() =>{
    var uid = "gbfBtH1XbDMYice2pM0zV7caEjn2";
    var pid = "0dyQVP3GOGDKeMxuEBuA";
    mockAddFavorite = jest.fn();
    wrapper = shallow(
        <StarButton  pid={pid} uid={uid} addFavorite={mockAddFavorite} />);   
});

it('should render starbutton correctly' , async () => {
   expect(wrapper).toMatchSnapshot();
})

it('should set starbutton state correctly' , async () => {  //tests that photobox is shown 
    expect(wrapper.state('favorite')).toBeFalsy();
 })

 it("Click", () => {
    const button = getByTestId(wrapper, 'button_fav');
    fireEvent.click(button);
});
