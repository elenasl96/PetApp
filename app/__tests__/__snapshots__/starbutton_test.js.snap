// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render starbutton correctly 1`] = `ShallowWrapper {}`;
